lib/rspec/rails/matchers/have_reported_error.rb

```diff
@@ -0,0 +1,271 @@
+module RSpec
+  module Rails
+    module Matchers
+      # @api private
+      # Sentinel value to distinguish between no argument passed vs explicitly passed nil.
+      # This follows the same pattern as RSpec's raise_error matcher.
+      UndefinedValue = Object.new.freeze
+
+      # @api private
+      class ErrorSubscriber
```

Comment:
They have https://github.com/rails/rails/blob/57093484e7926fb33ff266aa25b3c6241b235a68/activesupport/lib/active_support/testing/error_reporter_assertions.rb#L6, can we reuse it?